library/src/main/java/com/heartlandpaymentsystems/library/terminals/c2x/C2XDevice.java

```diff
@@ -206,7 +206,9 @@ public void setConnectionConfig(ConnectionConfig connectionConfig) {
         terminalConfig.setOutputCapability(TerminalOutputCapability.PRINT_AND_DISPLAY);
         terminalConfig.setAuthenticationCapability(TerminalAuthenticationCapability.NO_CAPABILITY);
         terminalConfig.setOperatingEnvironment(TerminalOperatingEnvironment.ON_MERCHANT_PREMISES_ATTENDED);
-        terminalConfig.setTimeout(Long.getLong(connectionConfig.getTimeout()));
+
+        final Long timeout = Long.getLong(connectionConfig.getTimeout());
+        terminalConfig.setTimeout(timeout != null ? timeout : 6000L);
```

Comment:
Is this in milliseconds? If so, 6 seconds would be too short. We'd want at least 60 seconds.

Reply:
I agree, I updated to 60000L. By the way, here's the crash logs on Crashlytics that this tries to fix:

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'long java.lang.Long.longValue()' on a null object reference
       at com.heartlandpaymentsystems.library.terminals.c2x.C2XDevice.setConnectionConfig(C2XDevice.java:209)
```